### PR TITLE
Extract language tags from FLEx projects and store in DB metadata

### LINKED ISSUE #922: Pull vernacular and analysis language from flex project data
this would be similar to the entries count data. We would store it in the flex metadata table, and we want to make it searchable/queryable in some way.

Tasks:
- [x] pull language data from xml files in hg
- [x] display language data on project page
- [x] allow filtering projects by language data via graphql

Non tasks:
- define UI in admin page to filter based on vernacular.

### PR DESCRIPTION
**Pull request by rmunn**
Fixes #922.

This PR extracts language tags from FLEx projects, but does nothing yet to validate or process them. That might come in a separate PR; this one just stores the language tag data into a JSON column and leaves the analysis for later.

### CODE REVIEW

File: backend/LexData/Migrations/LexBoxDbContextModelSnapshot.cs [hunk 1]
```diff
@@ -1164,6 +1164,75 @@ protected override void BuildModel(ModelBuilder modelBuilder)
...
+
+                            b1.ToJson("WritingSystems");
+
+                            b1.WithOwner()
+                                .HasForeignKey("FlexProjectMetadataProjectId");
+
+                            b1.OwnsMany("LexCore.Entities.FLExWsId", "AnalysisWss", b2 =>
+                                {
+                                    b2.Property<Guid>("ProjectWritingSystemsFlexProjectMetadataProjectId")
+                                        .HasColumnType("uuid");
```
Comment (rmunn) [inline backend/LexData/Migrations/LexBoxDbContextModelSnapshot.cs:1185]: Why is this here? What actual DB column does it correspond to? I have no idea why EF does this.

**Comment by rmunn** [comment]
Actually, now that I take another look at #922 this doesn't yet do everything the issue asks for. For example, there's no GraphQL filtering available yet. Marking as draft and removing review request for now.

**Review by rmunn — COMMENTED** [review]
Self-review finished. Marked a few places I'd like specific feedback on: e.g,, should I spend the time to do the proposed refactor, or is it readable enough already and I should move on to something else?

File: backend/LexBoxApi/Services/ProjectService.cs [hunk 2]
```diff
@@ -58,6 +58,29 @@ public async Task<Guid> CreateProject(CreateProjectInput input)
...
+
+    public async Task UpdateProjectLangProjectId(Guid projectId)
+    {
+        var project = await dbContext.Projects.FindAsync(projectId);
+        if (project is null || project.Type != ProjectType.FLEx) return;
+        await dbContext.Entry(project).Reference(p => p.FlexProjectMetadata).LoadAsync();
+        var langProjGuid = await hgService.GetProjectIdOfFlexProject(project.Code);
+        project.FlexProjectMetadata ??= new FlexProjectMetadata();
+        project.FlexProjectMetadata.LangProjectId = langProjGuid;
+        await dbContext.SaveChangesAsync();
```
[the remark this answers is not in the document]
Comment (hahn-kev) [reply backend/LexBoxApi/Services/ProjectService.cs:81]: I think in that case the caller can just call `IHgService.GetProjectIdOfFlexProject`

**Review by hahn-kev — CHANGES_REQUESTED** [review]
just some minor requests, including removing an empty migration

File: frontend/src/routes/(authenticated)/project/[project_code]/+page.svelte [hunk 3]
```diff
@@ -77,6 +80,11 @@
     loadingEntryCount = false;
   }
 
+  async function updateLanguageList(): Promise<void> {
+    await fetch(`/api/project/updateLanguageList/${project.code}`, {method: 'POST'});
+    await invalidate(`project:${project.code}`);
```
[the remark this answers is not in the document]
Comment (rmunn) [reply frontend/src/routes/(authenticated)/project/[project_code]/+page.svelte:85]: Kevin and I decided to change this to a mutation, but in a separate PR so as not to expand the scope of the current PR too much. I opened https://github.com/sillsdev/languageforge-lexbox/issues/970 to track that work.

File: backend/LexBoxApi/Services/HgService.cs [hunk 4]
```diff
@@ -160,6 +160,55 @@ private async Task CleanupRepoFolder(DirectoryInfo repoDir)
...
+        var root = doc.DocumentElement;
+        if (root is null) return null;
+        var vernWssStr = root["VernWss"]?["Uni"]?.InnerText ?? "";
+        var analysisWssStr = root["AnalysisWss"]?["Uni"]?.InnerText ?? "";
+        var curVernWssStr = root["CurVernWss"]?["Uni"]?.InnerText ?? "";
+        var curAnalysisWssStr = root["CurAnalysisWss"]?["Uni"]?.InnerText ?? "";
+        var vernWss = vernWssStr.Split((char[]?)null, StringSplitOptions.RemoveEmptyEntries);
+        var analysisWss = analysisWssStr.Split((char[]?)null, StringSplitOptions.RemoveEmptyEntries);
+        var curVernWss = curVernWssStr.Split((char[]?)null, StringSplitOptions.RemoveEmptyEntries);
+        var curAnalysisWss = curAnalysisWssStr.Split((char[]?)null, StringSplitOptions.RemoveEmptyEntries);
```
[the remark this answers is not in the document]
Comment (rmunn) [reply backend/LexBoxApi/Services/HgService.cs]: Done in commit 27699a05.

File: backend/LexData/Migrations/20240715044054_AddIsDefaultFieldToFLExWsId.cs
[the remark this answers is not in the document]
Comment (rmunn) [reply backend/LexData/Migrations/20240715044054_AddIsDefaultFieldToFLExWsId.cs:1]: Yeah, should be safe, Done in commit 3cbc6fda.

File: frontend/src/routes/(authenticated)/project/[project_code]/+page.svelte [hunk 5]
```diff
@@ -332,6 +340,34 @@
           </AdminContent>
         </DetailItem>
       {/if}
+      {#if project.type === ProjectType.FlEx}
+        <DetailItem title={$t('project_page.vernacular_langs')}>
+          <WritingSystemList writingSystems={vernacularLangTags} />
+          <AdminContent>
+            <IconButton
+              loading={loadingEntryCount}
```
[the remark this answers is not in the document]
Comment (rmunn) [reply frontend/src/routes/(authenticated)/project/[project_code]/+page.svelte]: Yep, bit of copy-and-paste laziness there. Fixed in commit 47117b4d.

File: frontend/src/routes/(authenticated)/project/[project_code]/+page.svelte [hunk 6]
```diff
@@ -332,6 +340,34 @@
...
+              outline={false}
+              on:click={updateLanguageList}
+            />
+          </AdminContent>
+        </DetailItem>
+        <DetailItem title={$t('project_page.analysis_langs')}>
+          <WritingSystemList writingSystems={analysisLangTags} />
+          <AdminContent>
+            <IconButton
+              loading={loadingEntryCount}
```
[the remark this answers is not in the document]
Comment (rmunn) [reply frontend/src/routes/(authenticated)/project/[project_code]/+page.svelte]: Fixed in commit 47117b4d.

**Review by hahn-kev — APPROVED** [review]
looks good to me

**Pull request merged**